lib/services/cordova-migration-service.ts

```diff
@@ -134,6 +143,163 @@ export class CordovaMigrationService implements ICordovaMigrationService {
+			}
+
+			_.each(this.invalidMarketplacePlugins, plugin => {
+				let {name} = this.$pluginsService.getPluginBasicInformation(plugin);
+				this.$pluginsService.removePlugin(name).wait();
+			});
+
+			_.each(this.$project.configurations, (configuration: string) => {
+				let oldPluginsList = this.$project.getProperty("CorePlugins", configuration);
+				let newPluginsList = this.migratePlugins(oldPluginsList, oldVersion, newVersion).wait();
+				this.$logger.trace("Migrated core plugins to: ", helpers.formatListOfNames(newPluginsList, "and"));
+				this.$project.setProperty("CorePlugins", newPluginsList, configuration);
+			});
+
+			this.migrateCordovaJsFiles(newVersion).wait();
+
+			this.$logger.info("Successfully migrated to version %s", versionDisplayName);
+		}).future<void>()();
+	}
+
+	public getSupportedPlugins(): IFuture<string[]> {
+		return (() => {
+			let version: string;
+			if(this.$project.projectData) {
+				version = this.$project.projectData.FrameworkVersion;
+			} else {
+				let selectedFramework = _.findLast(this.getSupportedFrameworks().wait(), (sv: Server.FrameworkVersion) => sv.DisplayName.indexOf(this.$projectConstants.EXPERIMENTAL_TAG) === -1);
+				version = selectedFramework.Version;
+			}
+
+			return this.pluginsForVersion(version).wait();
+		}).future<string[]>()();
+	}
+
+	private migrateCordovaJsFiles(newVersion: string): IFuture<void> {
+		return ((): void => {
+			let backedUpFiles: string[] = [],
+				backupSuffix = ".backup";
+			try {
+				_.each(this.$mobileHelper.platformNames, (platform) => {
+					this.$logger.trace("Replacing cordova.js file for %s platform ", platform);
+					let cordovaJsFileName = path.join(this.$project.getProjectDir().wait(), `cordova.${platform}.js`.toLowerCase());
+					let cordovaJsSourceFilePath = this.$resources.buildCordovaJsFilePath(newVersion, platform);
+					this.$fs.copyFile(cordovaJsFileName, cordovaJsFileName + backupSuffix).wait();
+					backedUpFiles.push(cordovaJsFileName);
+					this.$fs.copyFile(cordovaJsSourceFilePath, cordovaJsFileName).wait();
+				});
+			} catch(error) {
+				_.each(backedUpFiles, file => {
+					this.$logger.trace("Reverting %s", file);
+					this.$fs.copyFile(file + backupSuffix, file).wait();
+				});
+				this.$errors.failWithoutHelp(error.message);
+			}
+			finally {
+				_.each(backedUpFiles, file => {
+					this.$fs.deleteFile(file + backupSuffix).wait();
+				});
+			}
+		}).future<void>()();
+	}
+
+	private promptUserForInvalidPluginsAction(plugins: string[], toVersion: string): IFuture<void> {
+		return (() => {
+			let remove = `Remove those plugins from all configurations`,
+				cancel = 'Cancel Cordova migration',
+				pluginsString = plugins.length > 1 ? `plugins ${plugins.join(', ')} are` : `plugin ${plugins.join(', ')} is`,
+				choice = this.$prompter.promptForChoice(`The ${pluginsString} not supported for Cordova version ${toVersion} what do you want to do?`, [remove, cancel]).wait();
```

Comment:
Maybe you need "." before what do you want to do. This will make the question more readable.